src/blockchain/log-processors/category-aggregations.ts

```diff
@@ -0,0 +1,87 @@
+import * as Knex from "knex";
+import { ReportingState } from "../../types";
+
+interface MarketOpenInterestChangedParams {
+  db: Knex;
+  categoryName: string;
+  newOpenInterest: BigNumber;
+  oldOpenInterest: BigNumber;
+  reportingState: ReportingState;
+}
+
+interface MarketFinalizedParams {
+  db: Knex;
+  marketId: string;
+}
+
+export async function updateCategoryAggregationsOnMarketOpenInterestChanged(params: MarketOpenInterestChangedParams): Promise<void> {
+  const row: {
+    openInterest: BigNumber,
+    nonFinalizedOpenInterest: BigNumber,
+  } | undefined = await params.db.first(["openInterest", "nonFinalizedOpenInterest"])
+    .from("categories")
+    .where({ category: params.categoryName });
+  if (row === undefined) throw new Error(`No category found with name ${params.categoryName}`);
+
+  const oiDelta: BigNumber = params.newOpenInterest.minus(params.oldOpenInterest);
+
+  const updates: any = {
+    // 1. update category.openInterest to reflect change in market's OI
+    openInterest: row.openInterest.plus(oiDelta).toString(),
+  };
+
+  if (params.reportingState !== ReportingState.FINALIZED) {
+    // 2. update category.nonFinalizedOpenInterest to reflect change in non-finalized market's OI
+    updates.nonFinalizedOpenInterest = row.nonFinalizedOpenInterest.plus(oiDelta).toString();
+  }
+
+  await params.db("categories").update(updates).where({ category: params.categoryName });
+}
+
+// precondition: market reportingState == FINALIZED in the db
+export async function updateCategoryAggregationsOnMarketFinalized(params: MarketFinalizedParams): Promise<void> {
+  // 1. update category.nonFinalizedOpenInterest because a finalized market is newly excluded from nonFinalizedOpenInterest
+  const v = await marketFinalizedHelper(params.db, params.marketId);
```

Comment:
Is there a better than `v` we could use here?

Reply:
renamed to `data`, it's just a grab bag of data returned by helper, it's type more interesting than identifier. working in golang turned me into a fan of terse local identifiers :P, in some cases